pennylane/interfaces/execution.py

```diff
@@ -244,23 +242,7 @@ def wrapper(tapes: Sequence[QuantumTape], **kwargs):
                 # Tape exists within the cache, store the cached result
                 cached_results[i] = cache[hashes[i]]
 
-                # Introspect the set_shots decorator of the input function:
-                #   warn the user in case of finite shots with cached results
-                finite_shots = False
-
-                closure = inspect.getclosurevars(fn).nonlocals
-                if "original_fn" in closure:  # deal with expand_fn wrapper above
-                    closure = inspect.getclosurevars(closure["original_fn"]).nonlocals
-
-                # retrieve the captured context manager instance (for set_shots)
-                if "self" in closure and isinstance(closure["self"], _GeneratorContextManager):
-                    # retrieve the shots from the arguments or device instance
-                    if closure["self"].func.__name__ == "set_shots":
-                        dev, shots = closure["self"].args

```

Comment:
just to confirm, this device-swapping would be for an identical device, but with different shots? which doesn't matter anymore because we're using tape shots from here onward?

Reply:
All of this is just a fancy way of telling if the execution has finite shots.  Now we just query the tape to figure out if there's finite shots.